tests/sources/xml/test_oai_dc.py

```diff
@@ -54,26 +65,217 @@ def test_oaidctransform_with_all_fields_transforms_correctly():
 
 
-def test_oaidc_generic_date():
+def test_get_content_type_success(oai_dc_record_all_fields):
     source_records = OaiDc.parse_source_file(
-        f"{FIXTURES_PREFIX}/oaidc_record_valid_generic_date.xml"
+        "tests/fixtures/oai_dc/oaidc_record_all_fields.xml"
     )
-    transformer_instance = OaiDc("libguides", source_records)
-    xml = next(transformer_instance.source_records)
-    assert transformer_instance.get_dates("test_source_record_id", xml) == [
-        timdex.Date(kind="Unknown", note=None, range=None, value="2008-06-19T17:55:27")
+    assert OaiDc("cool-repo", source_records).get_content_type() == ["cool-repo"]
+
+
+def test_get_content_type_raises_key_error_if_source_blank(oai_dc_record_all_fields):
+    with pytest.raises(KeyError):
+        OaiDc("", oai_dc_record_all_fields).get_content_type()
+
+
+def test_get_content_type_raises_key_error_if_source_missing(oai_dc_record_all_fields):
+    with pytest.raises(KeyError):
+        OaiDc(None, oai_dc_record_all_fields).get_content_type()
+
+
+def test_get_contributors_success():
+    source_record = create_oai_dc_source_record_stub(
+        """
+        <dc:creator>Ye Li</dc:creator>
+        """
+    )
+    assert OaiDc.get_contributors(source_record) == [
+        timdex.Contributor(
+            value="Ye Li",
+            affiliation=None,
+            identifier=None,
+            kind="Creator",
+            mit_affiliated=None,
+        )
+    ]
+
+
+def test_get_contributors_transforms_correctly_if_fields_blank(
+    oai_dc_record_optional_fields_blank,
+):
+    assert OaiDc.get_contributors(oai_dc_record_optional_fields_blank) == []
+
+
+def test_get_contributors_transforms_correctly_if_fields_missing(
+    oai_dc_record_optional_fields_missing,
+):
+    assert OaiDc.get_contributors(oai_dc_record_optional_fields_missing) == []
+
+
+def test_get_dates_success():
+    source_record = create_oai_dc_source_record_stub(
+        """
+        <dc:date>2008-06-19T17:55:27</dc:date>
+        """
+    )
+    assert OaiDc.get_dates(source_record_id="abc", source_record=source_record) == [
+        timdex.Date(kind="Unknown", value="2008-06-19T17:55:27")
     ]
+
+
+def test_get_dates_transforms_correctly_if_fields_blank(
+    oai_dc_record_optional_fields_blank,
+):
+    assert (
+        OaiDc.get_dates(
+            source_record_id="abc", source_record=oai_dc_record_optional_fields_blank
+        )
+        == []
+    )
+
+
+def test_get_dates_transforms_correctly_if_fields_missing(
+    oai_dc_record_optional_fields_missing,
+):
+    assert (
+        OaiDc.get_dates(
+            source_record_id="abc", source_record=oai_dc_record_optional_fields_missing
+        )
+        == []
+    )
+
+
+def test_get_dates_transforms_correctly_if_date_invalid():
+    source_record = create_oai_dc_source_record_stub(
+        """
+        <dc:date>INVALID</dc:date>
+        """
+    )
+    assert OaiDc.get_dates(source_record_id="abc", source_record=source_record) == []
```

Comment:
This test replaces the old test removed via [this commit](https://github.com/MITLibraries/transmogrifier/pull/177/commits/cd5569a419521205d9f95f8977bdfdc1b0dec2e8).

Reply:
Just updated the hyperlink above. Thanks!